src/Geocoder/Formatter/StringFormatter.php

```diff
@@ -51,19 +49,26 @@ class StringFormatter
      */
     public function format(Address $address, $format)
     {
-        return strtr($format, array(
+        $tr = [
             self::STREET_NUMBER => $address->getStreetNumber(),
             self::STREET_NAME   => $address->getStreetName(),
             self::LOCALITY      => $address->getLocality(),
             self::POSTAL_CODE   => $address->getPostalCode(),
             self::SUB_LOCALITY  => $address->getSubLocality(),
-            self::COUNTY        => $address->getCounty()->getName(),
-            self::COUNTY_CODE   => $address->getCounty()->getCode(),
-            self::REGION        => $address->getRegion()->getName(),
-            self::REGION_CODE   => $address->getRegion()->getCode(),
             self::COUNTRY       => $address->getCountry()->getName(),
             self::COUNTRY_CODE  => $address->getCountry()->getCode(),
             self::TIMEZONE      => $address->getTimezone(),
-        ));
+        ];
+
+        $adminLevels = $address->getAdminLevels();
+        $nullAdminLevel = new AdminLevel(null, null, null);
```

Comment:
I don't understand what you mean, since the returned value is the formatted string.
Here I've used a null object just to avoid two cycle: the first to set all ADMIN_LEVEL_* `$tr` to null, and the second (foreach adminLevels) to set the actual value for the address. Since `AdminLevel` class is final there should be no problem.